lib/apiservers/engine/backends/backends.go

```diff
@@ -92,44 +93,89 @@ func Init(portLayerAddr, product string, config *config.VirtualContainerHostConf
 	// the vic-machine installer timeout will intervene if this blocks for too long
 	pingPortLayer()
 
+	if err := hydrateCaches(); err != nil {
+		return err
+	}
+
 	log.Info("Creating image store")
 	if err := createImageStore(); err != nil {
 		log.Errorf("Failed to create image store")
 		return err
 	}
 
+	serviceOptions := registry.ServiceOptions{}
+	for _, registry := range insecureRegs {
+		insecureRegistries = append(insecureRegistries, registry.Path)
+	}
+	if len(insecureRegistries) > 0 {
+		serviceOptions.InsecureRegistries = insecureRegistries
+	}
+	log.Debugf("New registry service with options %#v", serviceOptions)
+	RegistryService = registry.NewService(serviceOptions)
+
+	return nil
+}
+
+func hydrateCaches() error {
+
+	const waiters = 4
+
+	wg := sync.WaitGroup{}
+	wg.Add(waiters)
+	errors := make(chan error, waiters)
+
+	log.Info("Refreshing layer cache")
+	go func() {
+		defer wg.Done()
+		if err := imagec.NewLayerCache(portLayerClient); err != nil {
+			errors <- fmt.Errorf("Failed to update layer cache: %s", err)
+			return
+		}
+		log.Info("Layer cache updated successfully")
+		errors <- nil
+	}()
+
 	log.Info("Refreshing image cache")
 	go func() {
-		if err := cache.ImageCache().Update(portLayerClient); err != nil {
-			log.Warnf("Failed to refresh image cache: %s", err)
+		defer wg.Done()
+		if err := cache.NewImageCache(portLayerClient); err != nil {
+			errors <- fmt.Errorf("Failed to refresh image cache: %s", err)
 			return
 		}
 		log.Info("Image cache updated successfully")
+		errors <- nil
+	}()
+
+	log.Info("Refreshing repository cache")
+	go func() {
+		defer wg.Done()
+		if err := cache.NewRepositoryCache(portLayerClient); err != nil {
+			errors <- fmt.Errorf("Failed to create repository cache: %s", err.Error())
+			return
+		}
+		errors <- nil
+		log.Info("Repository cache updated successfully")
 	}()
 
 	log.Info("Refreshing container cache")
 	go func() {
+		defer wg.Done()
 		if err := syncContainerCache(); err != nil {
-			log.Warnf("Failed to refresh container cache: %s", err)
+			errors <- fmt.Errorf("Failed to refresh container cache: %s", err)
 			return
 		}
 		log.Info("Container cache updated successfully")
+		errors <- nil
 	}()
 
-	// creates and potentially restore repository cache
-	if err := cache.NewRepositoryCache(portLayerClient); err != nil {
-		return fmt.Errorf("Failed to create repository cache: %s", err.Error())
-	}
+	wg.Wait()
+	close(errors)
 
-	serviceOptions := registry.ServiceOptions{}
-	for _, registry := range insecureRegs {
-		insecureRegistries = append(insecureRegistries, registry.Path)
-	}
-	if len(insecureRegistries) > 0 {
-		serviceOptions.InsecureRegistries = insecureRegistries
+	for err := range errors {
+		if err != nil {
+			return err
```

Comment:
if you don't care too much about the error type, you can just append all error messages. otherwise, there might have only one error message log out